packages/components/src/number-control/README.md

```diff
@@ -97,6 +97,20 @@ The minimum `value` allowed.
 -   Required: No
 -   Default: `-Infinity`
 
+### onChange
+
+Callback fired whenever the value of the input changes.
+
+The callback receives two arguments:
+
+1. `newValue`: the new value of the input
+2. `extra`: an object containing, under the `event` key, the original browser event.
```

Comment:
I think I have a slight (though not at all strong) preference for having the second param be the plain event object rather than nesting it within an `extra` object. There's a familiarity with having the second param of callbacks be the event object itself, so it might make this component slightly easier for someone new to looking at it if we don't add in this abstraction.

But I'm aware you probably added it in mostly thinking of the `isValid` flag? I think given the documentation now covers `event.target.validity.valid`, we'd probably be fine not adding in any additional metadata.

What do you think?

Reply:
Chiming in to say the shape of that goes back well before this PR. I've suggested the same https://github.com/WordPress/gutenberg/pull/33696#discussion_r677835665. My feeling is the components are experimental so we have license to make the change as long as we fix up any internal stuff that might break. Good for a separate PR though.

Reply:
As @stokesman mentioned, the `extra` object existed before this PR and [is defined in `InputControl`](https://github.com/WordPress/gutenberg/blob/2df751aa2ad079b9c015bc159720d61b48780e0c/packages/components/src/input-control/types.ts#L35-L38).

For example, `UnitControl` [extends that `extra` object ](https://github.com/WordPress/gutenberg/blob/0ea49d674a376636db5c0722ae41083394df9e60/packages/components/src/unit-control/types.ts#L40-L43)by adding a custom `data` property too (which seems to include extra info about which unit is being selected).

Given the above, I'm not sure what would be the best way to simplify the signature of the `onChange` callback — but it's definitely something that can be done in a follow-up

Reply:
Ah, excellent, thanks for linking the prior discussion, glad I'm not alone there! 😄

Yes, I think that's perfectly fine for a follow-up, not a blocker for me!